### DISCUSSION TITLE
How to force next to re-execute page.tsx server component?

### DISCUSSION THREAD

**Discussion by vatosarmat** [vercel/next.js#66207] [Help]
### Summary

`page.tsx` is a server component, it fetches data from the database and pass it to client components as props. Sometimes old data from the previous execution comes. Seems like I need to invalidate some cache. How to tell `next`, that particular `page.tsx` on particular path must be re-executed? I've tried `revalidatePath` but with no success. Is `revalidatePath` for that and maybe I did something wrong?

**Comment by serdar-webdev** [comment]
Hi, revalidatePath we use to update fetchs in some server component, it means if you do some mutations to your data, for example: you have "users" fetch and it returns users, after you mutate this users you have to revalidatePath or revalidateTag ( of your fetch ), and it will re-generate this server component with fresh data. Can you provide code?
